### PR TITLE
Added new Authorization Roles

### PR DESCRIPTION
**Pull request by enrichman**
Update of the documentation matching the new Authorization Roles:

See: https://github.com/epinio/epinio/issues/2631
- ADR: https://github.com/epinio/epinio/pull/2677
- Epinio: https://github.com/epinio/epinio/pull/2656
- Charts: https://github.com/epinio/helm-charts/pull/506

### CODE REVIEW

File: docs/howtos/customization/create_custom_role.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+sidebar_label: "Creating A Git Configuration"
```
Comment (mmartin24) [inline docs/howtos/customization/create_custom_role.md:2]: To confirm, was it meant this or instead "Creating a custom role"?

Comment (enrichman) [reply docs/howtos/customization/create_custom_role.md:2]: Ah, yes. Copy-pasta :spaghetti:

File: docs/howtos/customization/create_custom_role.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  id: custom-role
+  name: "My Custom Role"
+  actions: |
+    app
+    configuration_read
+    service_read
+EOF
+```
+
+Then to assign the role to a user you can update the user `epinio.io/roles` annotation:
```
Comment (jhkrug) [inline docs/howtos/customization/create_custom_role.md:33]:
```suggestion
Then, to assign the role to a user you can update the user `epinio.io/roles` annotation:
```

File: docs/references/authentication_oidc.md [hunk 3]
```diff
@@ -71,4 +74,4 @@ config.yaml: |-
       - name: Org1
 ```
 
-The groups will be evaluated in order. In the previous example if the user is a member of both `Org1:Admins` and `Org1:TeamBlue` then that user will get the `admin` role.
+In the previous example if the user is a member of both `Org1:Admins` and `Org1:TeamBlue` then that user will get the `admin`, `user` and `admin::workspace` roles. If no roles will be found then the user will get the "default" Epinio role.
```
Comment (jhkrug) [inline docs/references/authentication_oidc.md:77]:
```suggestion
In the previous example if the user is a member of both `Org1:Admins` and `Org1:TeamBlue` then that user will get the `admin`, `user` and `admin::workspace` roles. If no roles are found then the user will get the "default" Epinio role.
```

File: docs/references/authorization.md [hunk 4]
```diff
@@ -5,9 +5,10 @@ title: ""
 
 # Authorization
 
-Since version **0.8.0** Epinio is shipped with an authorization layer recognizing two basic roles: **admin** and **user**.
-A user with the admin role will have access to every resource, while a standard user will have access only to the resources created on its namespaces.
-When a user creates a namespace, it will have automatically permission for it.
+Since version **1.11.0** Epinio is shipped with a new authorization layer with two default roles: **admin** and **user**.
+Roles can be "namescoped", and they can be also customized with different permissions.
+
+A user with the admin role will have the permission to perform any operation. A standard user will have only read permissions, but it also has te permissions to create namespaces. When a user creates a namespace, it will have automatically the admin permission for it.
```
Comment (jhkrug) [inline docs/references/authorization.md:11]:
```suggestion
A user with the admin role will have the permission to perform any operation. A standard user will have only read permissions, but also has the permissions to create namespaces. When a user creates a namespace, they will automatically have the admin permission for it.
```

File: docs/references/authorization.md [hunk 5]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+| default | (optional) if set to _true_ the role will be the one selected as default if no other roles were assigned to the user
+| actions | The actions the roles can perform
+
+## Actions
+
+Each Role can perform some actions defined in the __actions__ field (newline separated). These actions are hardcoded in Epinio, and some of them depends on other actions. Enabling an action with a dependency will enable also the dependency automatically.
+
+### Namespace
+
+These actions will enable operations on the Namespace commands and resources.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:139]:
```suggestion
These actions enable operations on Namespace commands and resources.
```

File: docs/references/authorization.md [hunk 6]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+
+| Action ID         | Description 
+|-------------------|-------------
+| `namespace_read`    | Read permissions (list, show)
+| `namespace_write`   | Write permissions (create, delete)<br/>Depends on: `namespace_read`
+| `namespace`         | All the above<br/>Depends on: `namespace_read`, `namespace_write`
+
+### App
+
+These actions will enable operations on the App commands and resources. They will also enable commands related to the AppChart (`epinio app chart`) and the application environment variables.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:149]:
```suggestion
These actions enable operations on App commands and resources. They also enable commands related to  AppCharts (`epinio app chart`) and application environment variables.
```

File: docs/references/authorization.md [hunk 7]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+| `app_read`        | Read permissions (app list and show, env list and show)
+| `app_logs`        | Read application logs
+| `app_write`       | Write permissions (app create, delete, push, export, stage, env set and unset)<br/>Depends on: `app_read`, `app_logs`
+| `app_exec`        | Perform an exec into a running application
+| `app_portforward` | Open a tunnel with the `port-forward` command
+| `app`             | All the above<br/>Depends on: `app_read`, `app_logs`, `app_write`, `app_exec`, `app_portforward`
+
+### Configuration
+
+These actions will enable operations on the Configuration commands and resources. Be aware that to bind a configuration you will still need the `app_write` permission.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:162]:
```suggestion
These actions enable operations on Configuration commands and resources. Be aware that to bind a configuration you still need the `app_write` permission as well.
```

File: docs/references/authorization.md [hunk 8]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+
+| Action ID           | Description 
+|----------------------|-------------
+| `configuration_read`  | Read permissions (list, show)
+| `configuration_write` | Write permissions (create, delete)<br/>Depends on: `configuration_read`
+| `configuration`       | All the above<br/>Depends on: `configuration_read`, `configuration_write`
+
+### Service
+
+These actions will enable operations on the Service commands and resources.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:173]:
```suggestion
These actions enable operations on Service commands and resources. 
```

File: docs/references/authorization.md [hunk 9]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+| Action ID             | Description 
+|-----------------------|-------------
+| `service_read`        | Read permissions (list, show)
+| `service_write`       | Write permissions (create, delete, bind, unbind)<br/>Depends on: `service_read`
+| `service_portforward` | Open a tunnel with the `port-forward` command
+| `service`             | All the above<br/>Depends on: `service_read`, `service_write`, `service_portforward`
+
+### Gitconfig
+
+These actions will enable operations on the Gitconfig commands and resources.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:184]:
```suggestion
These actions enable operations on Gitconfig commands and resources.
```

File: docs/references/authorization.md [hunk 10]
```diff
@@ -99,6 +95,109 @@ stringData:
...
+
+| Action ID         | Description 
+|-------------------|-------------
+| `gitconfig_read`    | Read permissions (list, show)
+| `gitconfig_write`   | Write permissions (create, delete)<br/>Depends on: `gitconfig_read`
+| `gitconfig`         | All the above<br/>Depends on: `gitconfig_read`, `gitconfig_write`
+
+### Export Registries
+
+This action will enable operations on the Export Registries commands and resources. Only read operations are available.
```
Comment (andreas-kupries) [inline docs/references/authorization.md:194]:
```suggestion
This action enable operations on Export Registries commands and resources. Only read operations are available.
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
Some additional tweaks beyond John's proposals.

**Review by jhkrug — APPROVED** [review]
lgtm

**Review by andreas-kupries — DISMISSED** [review]

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**